### PR TITLE
refactor(contract): Use address for authentication

### PR DESCRIPTION
**Pull request by howjmay**
Use Ethereum address for authentication during the interaction with
the contract can solve the dramatic vulnerability caused by calling
contract functions with secret key. And it somehow reduce the number
of function parameters, too.

### CODE REVIEW

File: contract/contracts/authRole.sol [hunk 1]
```diff
@@ -5,9 +5,11 @@ import "./auditLog.sol";
 
 contract Authentication {
     enum OP {SELECT, DELETE, UPDATE, CREATE, INSERT}
-    // An owner can take all the CRUD operations.
-    // A maintainer is only allowed to do INSERT, UPDATE and SELECT.
-    // A viewer can only take SELECT operation.
+    /* 
+     * OWNER: Owners can take all the CRUD operations.
+     * MAINTAINER: Maintainers are only allowed to do INSERT, UPDATE and SELECT.
+     * VIEWER: Viewers can only take SELECT operation.
```
Comment (jserv) [inline contract/contracts/authRole.sol]: Should be "execute" SELECT operations.

File: contract/test/authRole_test.js [hunk 2]
```diff
@@ -64,93 +56,89 @@ contract("authRole", function (accounts) {
...
-      skMaintainerTest,
+      accounts[1],
       tableNameTest,
       maintainerRole
     );
-    assert.equal(authGrantPermission, true, "Assign skTest as MAINTAINER");
+    assert.equal(
+      authGrantPermission,
+      true,
+      "Assign `accounts[1]` as a MAINTAINER"
```
Comment (jserv) [inline contract/test/authRole_test.js]: Use "grant"

Comment (jserv) [reply contract/test/authRole_test.js]: `accounts[1]` might be inconvenient. How about introducing mnemonic terms?

Comment (howjmay) [reply contract/test/authRole_test.js]: How about using `a given address` to replace `accounts[1]`?

Comment (jserv) [reply contract/test/authRole_test.js]: It sounds good. You only have to mention the exact address in the very beginning.

Comment (howjmay) [reply contract/test/authRole_test.js]: `accounts[]` is a list of ethereum addresses returned by the test framework. The addresses are generated every time we run the test, so actually, we have no idea what the address exact is

Comment (jserv) [reply contract/test/authRole_test.js]: You can still prompt the value of `accounts` at first glance. Then, we don't have to embed its value later.

Comment (howjmay) [reply contract/test/authRole_test.js]: OK

**Pull request merged**